feat(router): BREAKING - router.push now uses PUSH navigation (#24600)
expo/expo: docs/pages/router/navigating-pages.mdx

}
```

## Understanding native navigation

> Expo Router employs a stack-based navigation approach. This means that each new route you navigate to is added to a stack. If you navigate to a route that's already in the stack, the stack is unwound back to that existing route.

Reviewer: ```suggestion
Expo Router uses a stack-based navigation approach. Each new route you navigate to gets added to a stack. If you navigate a route already in the stack, the stack unwinds back to that existing route.
```

- Let's keep this in present tense for clairty
- Also, let's be a bit concise

> For example, if you navigate from `/feed` to `/profile`, the stack will contain `/feed` and `/profile`. If you then navigate to `/settings`, the stack will contain `/feed`, `/profile`, and `/settings`. If you then navigate to `/feed`, the stack will unwind back to `/feed`.

Reviewer: ```suggestion
For example, when you navigate from `/feed` to `/profile`, the stack contains `/feed` and `/profile`. If you then navigate to `/settings`, the stack contains `/feed`, `/profile`, and `/settings`. If you then navigate back to `/feed`, the stack unwinds back to `/feed`.
```

> If you wish to navigate to a route without the stack unwinding, you can use the `push` prop on the `<Link>` component. This will always push the route onto the stack, even if it already exists.

Reviewer: ```suggestion
To navigate to a route without the stack unwinding, you can use the `push` prop on the `<Link>` component. This alwayss pushes the route onto the stack, even if it already exists.
```

In contrast, the `replace` method substitutes the current route in the navigation stack with a new one, effectively replacing the current screen with the new one without adding to the stack.

> 

Reviewer: ```suggestion
```

The `router` object is immutable and contains the following functions:

> - **naviate**: `(href: Href) => void` Navigate to a route using the default behaviour. You can provide a full path like **/profile/settings** or a relative path like **../settings**. Navigate to dynamic routes by passing an object like `{ pathname: 'profile', params: { id: '123' } }`.

Reviewer: ```suggestion
- **navigate**: `(href: Href) => void` Navigate to a route using the default behavior. You can provide a full path like **/profile/settings** or a relative path like **../settings**. Navigate to dynamic routes by passing an object like `{ pathname: 'profile', params: { id: '123' } }`.
```

- Typo and we prefer American English for spellings

}
```

## Pushing screens

By default, links `navigate` to the nearest route in the navigation stack, either by pushing a new route or unwinding to an existing route. You can use the `push` prop to always push the route onto the stack.

{/* prettier-ignore */}

> ```js app/index.js

Reviewer 2: ```suggestion
```jsx app/index.js
```

Syntax highlight nit.

}
```

## Pushing screens

> By default, links "push" new routes onto the navigation stack or unwind to a existing route. You can use the `push` prop to always push the route onto the stack.

Reviewer 3: ```suggestion
By default, links `navigate` to the nearest route in the navigation stack, either by pushing a new route or unwinding to an existing route. You can use the `push` prop to always push the route onto the stack.
```
